nestadia/src/lib.rs

```diff
@@ -142,37 +147,113 @@ impl Emulator {
     pub fn cpu(&self) -> &Cpu {
         &self.cpu
     }
-}
 
-pub fn frame_to_rgb(frame: &PpuFrame, output: &mut [u8; 256 * 240 * 3]) {
-    for i in 0..frame.len() {
-        let f = RGB_PALETTE[(frame[i] & 0x3f) as usize];
-        output[i * 3] = f[0];
-        output[i * 3 + 1] = f[1];
-        output[i * 3 + 2] = f[2];
+    pub fn frame_to_rgb(mask_reg: MaskReg, frame: &PpuFrame, output: &mut [u8; 256 * 240 * 3]) {
```

Comment:
Why do you take mask_reg as an argument instead of using &self and using it from there? It would make it easier to use from the front-ends I think?

Reply:
I just left it a static function because that's what it was to begin with. Also, it means another borrow and I thought that getting the mask_reg ourself gave us more control over when we borrowed it.

Reply:
I tested for a bit and it's not an "easy fix" because the frame is a mutable reference. We might fix it later if we change the frame buffer to be a boxed array instead, but for now we can leave it as is.